src/extension.ts

```diff
@@ -103,6 +111,64 @@ ${prompt}`;
 	return {};
 };
 
+async function handleCodeFlowVisualization(extContext: vscode.ExtensionContext, request: vscode.ChatAgentRequest, context: vscode.ChatAgentContext, progress: vscode.Progress<vscode.ChatAgentProgress>, token: vscode.CancellationToken): Promise<vscode.ChatAgentResult2> {
+	const editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		return {};
+	}
+
+	const selection = editor.selection;
+	if (selection.isEmpty) {
+		return {};
+	}
+
+	const document = editor.document;
+	const text = document.getText(selection);
+
+	const imageGenPrompt = `Generate a whiteboard-style flowchart that visually represents the following piece of code. The flowchart should clearly illustrate the logic and flow of the code, including decision points, loops, and function calls. Use standard flowchart symbols: rectangles for operations or assignments, diamonds for decision points, ovals for start and end points, and arrows to show the flow direction. Label each symbol with brief descriptions corresponding to the code. The code consists of a simple 'if-else' statement, a 'for' loop, and a function call. Ensure the flowchart is neatly organized, easy to read, and accurately reflects the sequence and conditions in the code. Use a clean, professional style suitable for a technical presentation.
+	
+	# Code:
+	\`\`\`
+	${text}
+	\`\`\``;
+	const { smallFilePath, resultUrl } = await getAiImage(extContext, imageGenPrompt);
+	const content = `![image](file://${smallFilePath})
+
+[Full size](${resultUrl})`;
+	progress.report({ content });
+
+	return {};
+};
+
+async function handleRender(extContext: vscode.ExtensionContext, request: vscode.ChatAgentRequest, context: vscode.ChatAgentContext, progress: vscode.Progress<vscode.ChatAgentProgress>, token: vscode.CancellationToken): Promise<vscode.ChatAgentResult2> {
+	const editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		return {};
+	}
+
+	const selection = editor.selection;
+	if (selection.isEmpty) {
+		return {};
+	}
+
+	const document = editor.document;
+	const text = document.getText(selection);
+
+	const imageGenPrompt = `Generate a high-resolution image of a website preview, based on the provided code. The image should display a realistic rendering of what the website would look like, including layout, colors, and typography. Emphasize accuracy in translating the code into a visual representation.
```

Comment:
You might get better results if you first ask copilot to generate a description of the rendered page from the code, then feed that description into dall-e